pkg/image/image.go

```diff
@@ -31,3 +39,26 @@ func GetTagsFromImage(image string) ([]string, error) {
 	tags, err := remote.List(repo, remote.WithAuthFromKeychain(authn.DefaultKeychain))
 	return tags, err
 }
+
+/* ParseReference is a wrapper function of imagesource.ParseReference
+   It provides support for oci: prefixes
+*/
+func ParseReference(ref string) (imagesource.TypedImageReference, error) {
+	dstType := imagesource.DestinationRegistry
+	switch {
```

Comment:
We are mirroring same conditionals in here https://github.com/openshift/oc/blob/b6f6c9ed655b1c60b636d524372380d55b681af0/pkg/cli/image/imagesource/reference.go#L98. When new case needs to be added, this function should be maintained as well. 

Instead of this, maybe we can pass all other conditions whose are not including `oci:` prefix to the main function. Does that simplify this function?. I mean something like this;

```go
if !strings.HasPrefix(ref, "oci:") {
return imagesource.ParseReference(ref)
}

dstType = DestinationOCI
ref = strings.TrimPrefix(ref, "oci:")
ref = strings.TrimPrefix(ref, "//") //it could be that there is none
ref = strings.TrimPrefix(ref, "/")  // case of full path

dst, err := libgoref.Parse(ref)
if err != nil {
       return imagesource.TypedImageReference{Ref: dst, Type: dstType}, fmt.Errorf("%q is not a valid image reference: %v", ref, err)
}
return imagesource.TypedImageReference{Ref: dst, Type: dstType}, nil
```

Reply:
Good idea. thnaks @ardaguclu . 
Commit updated.